src/sentry/api/endpoints/monitor_checkin_details.py

```diff
@@ -180,6 +184,15 @@ def put(self, request: Request, project, monitor, checkin) -> Response:
             duration = int((current_datetime - checkin.date_added).total_seconds() * 1000)
             params["duration"] = duration
 
+        if "attachment" in result:
```

Comment:
I'm wondering if we should store a `attachment_type` as well. We probably want to look more into how event's handle attachments?

Reply:
yup I can accept it in the API and store it in the `File` column using a content-type header similar to how it works with event attachments. Though now that I understand event attachments better I'm not even sure if this is what we should be mimicking.

If we choose to mimic event attachments we would have a user experience similar to this
https://docs.sentry.io/platforms/javascript/enriching-events/attachments/

But I'm wondering if users just want to send small bits of json data and see them rendered in the frontend. For example one user wrote in just wanting to track number of customers processed on every run and probably track that at a glance in their dashboard. In that case should we just store some kind of dictionary field and render it to the frontend similar to how we show additional data/contexts in issue details?

Reply:
I guess alternatively this could be two separate use cases, sometimes users will want to send smaller-form context data and sometimes they will want to send whole files like how event attachments are, but I feel like the former would be much more useful at the moment.